File: testdata/jennies/rawtypes/constant_reference_discriminator/JavaRawTypes/constant_reference_discriminator/GridLayoutUsingValue.java
package constant_reference_discriminator;

import java.util.Objects;

public class GridLayoutUsingValue {
    public String kind;
    public String gridLayoutProperty;
    public GridLayoutUsingValue() {
        this.kind = Constants.GridLayoutKindType;
        this.gridLayoutProperty = "";
    }
    public GridLayoutUsingValue(String gridLayoutProperty) {
        this.kind = Constants.GridLayoutKindType;
        this.gridLayoutProperty = gridLayoutProperty;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof GridLayoutUsingValue)) return false;
        GridLayoutUsingValue o = (GridLayoutUsingValue) other;
        if (!Objects.equals(this.kind, o.kind)) return false;
        if (!Objects.equals(this.gridLayoutProperty, o.gridLayoutProperty)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.kind, this.gridLayoutProperty);
    }
}

File: testdata/jennies/rawtypes/constant_reference_discriminator/CSharpRawTypes/src/Grafana/Foundation/ConstantReferenceDiscriminator/GridLayoutUsingValue.cs
namespace Grafana.Foundation.ConstantReferenceDiscriminator;


public class GridLayoutUsingValue
{
    public string Kind;
    public string GridLayoutProperty;

    public GridLayoutUsingValue()
    {
        this.Kind = default!;
        this.GridLayoutProperty = "";
    }

    public GridLayoutUsingValue(string kind, string gridLayoutProperty)
    {
        this.Kind = kind;
        this.GridLayoutProperty = gridLayoutProperty;
    }
}
